### DISCUSSION TITLE
Wishing for some explanation for 'Symbol' in oxc_semantic

### DISCUSSION THREAD

**Discussion by Dangerise** [oxc-project/oxc#2078] [Q&A] [answered]
I'm a little puzzled about the usage of the oxc_semantic crate, so I'm longing for more help.

Maybe it's necessary to complete the contribution guide.

**Comment by Boshen** [comment, marked as answer]
The crate mimics the binding phase from TypeScript https://github.com/microsoft/TypeScript/blob/main/src/compiler/binder.ts


It sets up and scopes and declarations. The playground has a scope and symbol tab on the top right corner. https://oxc-project.github.io/oxc/playground/

You may be more familiar with the term "symbol table". https://en.wikipedia.org/wiki/Symbol_table

**Comment by Dangerise** [comment]
Thanks, and it's much clear for me now.
